### PR TITLE
Refactor .taskcluster.yml, add tests

### PR DESCRIPTION
**Pull request by Pike**
Refactoring `.taskcluster.yml` to be a bit more friendly to make decisions based on `tasks_for`.

Also, make sure we're only running tc on push to master.

And add tests to ensure that stays that way.

### CODE REVIEW

**Comment by Pike** [comment]
@tomprince, this one actually needs review.

Not sure if the refactors in `.taskcluster.yml` go into the right direction, and such. Also, the tests on that need a design review, not just a code review. I took much of the inspiration from Marco's validator, but go a step further in that I inspect the tasks to match the input.

If this is OK, I'd do tests for PRs in a follow-up (also making sure we don't rerun tests on review requests, for example)

File: tests/tc/test_taskcluster_yml.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
```
Comment (tomprince) [inline tests/tc/test_taskcluster_yml.py:3]: There exists https://github.com/marco-c/taskcluster_yml_validator which attempts to do something similar to this. (I don't know any more about it, than it exists)

**Review by tomprince — APPROVED** [review]
Sorry for the delay in revewing this. It looks reasonable. The skeleton of this `.taskcluster.yml` is used in ~ all of our taskgraph-using github projects, though they have all divereged somewhat :frowning:, so it would be good to keep it close, but this does appear to do that (after taking into account whitespace).

File: tests/tc/test_taskcluster_yml.py (hunk 1 above)
Comment (Pike) [reply tests/tc/test_taskcluster_yml.py:3]: Yeah, I've stolen most of this from the validator. It doesn't expose the data it tests, so the follow-up tests I'm doing to ensure that we're not triggering decisions tasks in some conditions can't be easily integrated there.

**Pull request merged**